### PR TITLE
Switch to compose state and kotlin flows - 1. iteration

### PR DESCRIPTION
**Pull request by ArnyminerZ**
I've added a test for `Settings`, though if you don't find it necessary we can get rid of it.

### CODE REVIEW

**Review by sunkup — CHANGES_REQUESTED** [review]
Please go ahead and make the functions suspendable :)

File: app/src/androidTest/java/at/bitfire/icsdroid/TestSettings.kt [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+        settings.forceDarkMode(true)
+        // Wait for the flow to update, or throw timeout
+        runBlocking {
+            withTimeout(1000) {
+                while (!forceDarkMode) {
+                    delay(10)
+                }
+            }
+        }
+        assert(forceDarkMode)
```
Comment (sunkup) [inline app/src/androidTest/java/at/bitfire/icsdroid/TestSettings.kt:38]: I believe Ricki prefers to use only junit framework assert statements. So not kotlin `assert()`, but only `assertTrue()`, `assertEquals()`, etc ...

**Review by sunkup — APPROVED** [review]
Nice. More tests are always good :+1: 
I just removed the `data.close()` calls. It says in the Kdoc, they don't do anything for `InputStream`.

I would like to try my hands on some of the new flow concepts as well with the 2. iteration, if that's okay with you?

**Comment by ArnyminerZ** [comment]
Yup, let's merge this one and you can work on 2

**Pull request merged**